### ISSUE TITLE
[feature request (`satisfies`)]: support ranges with `||`

### ISSUE THREAD

**Issue by kindoflew** [omichelsen/compare-versions#63]
Hi, thanks for this library!

I was wondering if it'd be possible to add support for ranges that contain multiple ranges using `||` -- ie, something like:

```
^12.22.0 || ^14.17.0 || >=16.0.0
```

I'm currently using a workaround where I split the string and test if any of them pass `satisfies`:

```javascript
const rangeArray = dep.version.split('||').map((range) => range.trim());
const doesSatisfy = rangeArray.some((range) => satisfies(version, range));
```

but I think it would be nice if it was supported out of the box. I'd be happy to open a PR if you are into this idea! Thanks again!

**Comment by omichelsen** [comment]
Thanks for suggesting this. I am working on adding this to the existing `satisfies` function so it adheres to the npm spec. I expect to release this as a breaking version v6, since it would be looking for spaces and potentially break existing usage.

I have released `v6.0.0-rc.1`, please check if it works as expected.

Example usage: `1.2.7 || >=1.2.9 <2.0.0`
Spec: https://docs.npmjs.com/cli/v6/using-npm/semver#ranges

**Comment by kindoflew** [comment]
I think I might have found a small bug in `satisfies` (I actually don't think it's related to this feature, but I noticed it while testing `v6.0.0-rc.1`):

A package in the project I'm using this for has an incorrectly formatted range (`>= 12.0.0` -- note the space between the operator and the version number). This seems to cause `satisfies` to try to parse `>=` by itself, which isn't a valid range, so it throws.

Not sure if this is a thing that should be resolved in userland or handled by the library. In my use case, I'm fetching these ranges from `npm` (reading the `engines` field of project dependencies), so I don't have control over how they're formatted.

However, if I do something like `satisfies(version, range.replaceAll(' ', ''))`, everything works as expected!

**Comment by omichelsen** [comment]
Thanks for testing. This is tricky, because in semver ranges `||` is an "or" and space is "and", so the simple solution of removing space would not work. It also throws in v5, but the message is a little clearer since it doesn't try to break up the statement. I'll see what the semver package does in this case and think about a solution.

**Comment by kindoflew** [comment]
Ohhh yeah, that's a great point.

**Comment by kindoflew** [comment]
It looks like they have this RegEx:
https://github.com/npm/node-semver/blob/3a8a4309ae986c1967b3073ba88c9e69433d44cb/internal/re.js#L158

which gets used here:
https://github.com/npm/node-semver/blob/3a8a4309ae986c1967b3073ba88c9e69433d44cb/classes/range.js#L97

**Comment by omichelsen** [comment]
I finally got around to working on this and released 6.0.0-rc.3 to address the malformed input for ranges (extra spaces).

Example: `> 1.2.3 <=  1.2.5   ||   0.0.0` => `>1.2.3 <=1.2.5 || 0.0.0`

Thanks for the feedback and suggestions!
